zcash_primitives/src/sapling/note_encryption.rs

```diff
@@ -215,33 +198,30 @@ impl<P: consensus::Parameters> Domain for SaplingDomain<P> {
         note: &Self::Note,
         esk: &Self::EphemeralSecretKey,
     ) -> Self::EphemeralPublicKey {
-        // epk is an element of jubjub's prime-order subgroup,
-        // but Self::EphemeralPublicKey is a full group element
-        // for efficiency of encryption. The conversion here is fine
-        // because the output of this function is only used for
-        // encoding and the byte encoding is unaffected by the conversion.
-        (note.g_d * esk).into()
+        EphemeralSecretKey(*esk)
+            .derive_public(note.g_d.into())
```

Comment:
This ends up calling [`ka_sapling_derive_public`](https://github.com/zcash/librustzcash/blob/19d8e2125e03df88783b23ae6dcad0ae69436e97/zcash_primitives/src/sapling/spec.rs#L83) which uses `PreparedBase` and `PreparedScalar`. Remind me, is it efficient to do that even for a single scalar mul with fresh base and scalar?

Reply:
Yes: #332.